main.c

```diff
@@ -229,6 +236,70 @@ static int child(int sock, pid_t target_pid, bool do_config_network,
     return 0;
 }
 
+/*
+ * Child (--target-type=bess)
+ *
+ * FIXME: We do not really need to fork the child for BESS mode
+ */
+static int child_bess(int sock, const char *bess_socket)
+{
+    int bess_listenfd = -1, bess_acceptfd = -1;
+    struct sockaddr_un bess_listenaddr, bess_acceptaddr;
+    socklen_t bess_acceptaddrlen = sizeof(struct sockaddr_un);
+    memset(&bess_acceptaddr, 0, sizeof(bess_acceptaddr));
+    memset(&bess_listenaddr, 0, sizeof(bess_listenaddr));
+
+    /* Listen on the BESS socket */
+    if ((bess_listenfd = socket(AF_UNIX, SOCK_SEQPACKET, 0)) == -1) {
+        perror("child_bess: socket");
+        goto error;
+    }
+    bess_listenaddr.sun_family = AF_UNIX;
+    if (bess_socket == NULL) {
+        fprintf(stderr, "the specified BESS socket path is NULL\n");
+        goto error;
+    }
+    if (strlen(bess_socket) >= sizeof(bess_listenaddr.sun_path)) {
+        fprintf(stderr, "the specified BESS socket path is too long (>= %lu)\n",
+                sizeof(bess_listenaddr.sun_path));
+        goto error;
+    }
+    strncpy(bess_listenaddr.sun_path, bess_socket,
+            sizeof(bess_listenaddr.sun_path) - 1);
+    unlink(bess_socket); /* avoid EADDRINUSE */
+    if (bind(bess_listenfd, (struct sockaddr *)&bess_listenaddr,
+             sizeof(bess_listenaddr)) < 0) {
+        perror("child_bess: bind");
+        goto error;
+    }
+    if (listen(bess_listenfd, 0) < 0) {
+        perror("child_bess: listen");
+        goto error;
+    }
+
+    /* Accept a connection, and send its connection to the parent */
+    printf("Waiting for connection to %s\n", bess_socket);
```

Comment:
we are using both stdout and stderr for diagnostic messages.  Should we just stick to stderr?

Reply:
fixed